### DISCUSSION TITLE
How to print assert condition in normal function?

### DISCUSSION THREAD

**Discussion by fly50789** [pytest-dev/pytest#10730] [Q&A]
In normal case

assert a>b,'assert message'

will not show variable a and b.

But in pytest, I can see variable when assert fail.

I wonder if I can see variable in normal function?

Or I shouldn't use that feature?

**Comment by RonnyPfannschmidt** [comment]
It's not clear what you mean by "see variable" and "normal function"

Can you provide an example and/or elaborate in more detail

**Reply by fly50789** [reply]
In normal case.

I cant see variable a,b in stderr
```
>>> a =1
>>> b =2
>>> assert a>b, 'assert message'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError: assert message
```
But in pytest, I can saw the result about variable a and b.

```
def test_assert():
    a=1
    b=3
    assert a>b,'assert message'
```

```
tests/remote/test_scheduler_v2.py::test_assert FAILED

================================================================================== FAILURES ==================================================================================
________________________________________________________________________________ test_assert _________________________________________________________________________________ 

    def test_assert():
        a=1
        b=3
>       assert a>b,'assert message'
E       assert message
E       assert 1 > 3
```

**Reply by fly50789** [reply]
Also sorry about I cant represent my opinion much clearer  since eng is not my mother tongue

**Comment by fly50789** [comment]
I know I can write code like below.
But I feel if it can add it automatically is very helpful
```
a= 1 
b = 2
a >b , f'{a=} {b=} assert fail'
```

**Comment by RonnyPfannschmidt** [comment]
this is the pytest assertion rewriting feature

i'm not aware of a implementation of it for normal python runnin

**Reply by fly50789** [reply]
@RonnyPfannschmidt  ok, thank you anyway.
Pytest help me a lot.
Thanks for this perfect tool.
